### PR TITLE
block_iscsi_lvm: New case to install on lvm device with iSCSI target

### PR DESCRIPTION
**Pull request by YongxueHong**
New a case that installs the guest OS on the lvm device which is
created on an iSCSI target.

ID: 1731616
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
@lolyu  @qingwangrh @zhenyzha @peixiu pls help me to review it, thanks

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=block_iscsi_lvm --guestname=RHEL.8.2 --clone=no
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks: PASS (904.95 s)

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.block_iscsi_lvm.with_
installation.cdrom.extra_cdrom_ks: ERROR: Command 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdk' failed.\nstdout: b"\nWelcome to fdisk (u
til-linux 2.32.1).\nChanges will remain in memory only, until you decide to write them.\nBe careful before using the write command.\n\nD
evice does not contain a r... (563.80 s)

The specific log update in bugzilla

**Comment by YongxueHong** [comment]
> (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.block_iscsi_lvm.with_
> installation.cdrom.extra_cdrom_ks: ERROR: Command 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdk' failed.\nstdout: b"\nWelcome to fdisk (u
> til-linux 2.32.1).\nChanges will remain in memory only, until you decide to write them.\nBe careful before using the write command.\n\nD
> evice does not contain a r... (563.80 s)
> 
> The specific log update in bugzilla

Hi @zhenyzha 
Have fixed it, please help me review it again, thanks.

**Comment by zhenyzha** [comment]
python3 -blockdev
 (1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks: PASS (2012.77 s)

python -drive
 (3/3) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.ppc64le.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks: PASS (2234.91 s)

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Comment by peixiu** [comment]
Tested on windows 2012-r2 failed when windows setup, hit error message as:
"Windows cannot be installed to this hard disk space. This hard disk space is formatted with an unsupported version of the NTFS file system...."

Check the log file found, create lvm with a "Linux filesystem" type, it should not compatibility with Windows installation used NTFS type.
 
21:03:00 INFO | Running 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdb'
21:03:00 DEBUG| [stdout]
21:03:00 DEBUG| [stdout] Welcome to fdisk (util-linux 2.32.1).
21:03:00 DEBUG| [stdout] Changes will remain in memory only, until you decide to write them.
21:03:00 DEBUG| [stdout] Be careful before using the write command.
21:03:00 DEBUG| [stdout]
21:03:00 DEBUG| [stdout] Device does not contain a recognized partition table.
21:03:00 DEBUG| [stdout] Created a new DOS disklabel with disk identifier 0x7a539419.
21:03:00 DEBUG| [stdout]
21:03:00 DEBUG| [stdout] Command (m for help): Created a new GPT disklabel (GUID: 352C6F1B-E8DA-854B-A8B8-DA1DAB910F23).
21:03:00 DEBUG| [stdout]
21:03:00 DEBUG| [stdout] Command (m for help): Partition number (1-128, default 1): First sector (16384-115343326, default 16384): Last sector, +sectors or +size{K,M,G,T,P} (16384-115343326, default 115343326):
21:03:00 DEBUG| [stdout] Created a new partition 1 of type 'Linux filesystem' and of size 55 GiB.
21:03:00 DEBUG| [stdout]
21:03:00 DEBUG| [stdout] Command (m for help): The partition table has been altered.
21:03:00 DEBUG| [stdout] Calling ioctl() to re-read partition table.
21:03:00 DEBUG| [stdout] Syncing disks.

**Comment by YongxueHong** [comment]
> Tested on windows 2012-r2 failed when windows setup, hit error message as:
> "Windows cannot be installed to this hard disk space. This hard disk space is formatted with an unsupported version of the NTFS file system...."
> 
> Check the log file found, create lvm with a "Linux filesystem" type, it should not compatibility with Windows installation used NTFS type.
> 
> 21:03:00 INFO | Running 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdb'
> 21:03:00 DEBUG| [stdout]
> 21:03:00 DEBUG| [stdout] Welcome to fdisk (util-linux 2.32.1).
> 21:03:00 DEBUG| [stdout] Changes will remain in memory only, until you decide to write them.
> 21:03:00 DEBUG| [stdout] Be careful before using the write command.
> 21:03:00 DEBUG| [stdout]
> 21:03:00 DEBUG| [stdout] Device does not contain a recognized partition table.
> 21:03:00 DEBUG| [stdout] Created a new DOS disklabel with disk identifier 0x7a539419.
> 21:03:00 DEBUG| [stdout]
> 21:03:00 DEBUG| [stdout] Command (m for help): Created a new GPT disklabel (GUID: 352C6F1B-E8DA-854B-A8B8-DA1DAB910F23).
> 21:03:00 DEBUG| [stdout]
> 21:03:00 DEBUG| [stdout] Command (m for help): Partition number (1-128, default 1): First sector (16384-115343326, default 16384): Last sector, +sectors or +size{K,M,G,T,P} (16384-115343326, default 115343326):
> 21:03:00 DEBUG| [stdout] Created a new partition 1 of type 'Linux filesystem' and of size 55 GiB.
> 21:03:00 DEBUG| [stdout]
> 21:03:00 DEBUG| [stdout] Command (m for help): The partition table has been altered.
> 21:03:00 DEBUG| [stdout] Calling ioctl() to re-read partition table.
> 21:03:00 DEBUG| [stdout] Syncing disks.

Hi @peixiu , I think this issue is related with https://github.com/autotest/tp-qemu/pull/2025#issuecomment-589471956 .

**Comment by peixiu** [comment]
> > Tested on windows 2012-r2 failed when windows setup, hit error message as:
> > "Windows cannot be installed to this hard disk space. This hard disk space is formatted with an unsupported version of the NTFS file system...."
> > Check the log file found, create lvm with a "Linux filesystem" type, it should not compatibility with Windows installation used NTFS type.
> > 21:03:00 INFO | Running 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdb'
> > 21:03:00 DEBUG| [stdout]
> > 21:03:00 DEBUG| [stdout] Welcome to fdisk (util-linux 2.32.1).
> > 21:03:00 DEBUG| [stdout] Changes will remain in memory only, until you decide to write them.
> > 21:03:00 DEBUG| [stdout] Be careful before using the write command.
> > 21:03:00 DEBUG| [stdout]
> > 21:03:00 DEBUG| [stdout] Device does not contain a recognized partition table.
> > 21:03:00 DEBUG| [stdout] Created a new DOS disklabel with disk identifier 0x7a539419.
> > 21:03:00 DEBUG| [stdout]
> > 21:03:00 DEBUG| [stdout] Command (m for help): Created a new GPT disklabel (GUID: 352C6F1B-E8DA-854B-A8B8-DA1DAB910F23).
> > 21:03:00 DEBUG| [stdout]
> > 21:03:00 DEBUG| [stdout] Command (m for help): Partition number (1-128, default 1): First sector (16384-115343326, default 16384): Last sector, +sectors or +size{K,M,G,T,P} (16384-115343326, default 115343326):
> > 21:03:00 DEBUG| [stdout] Created a new partition 1 of type 'Linux filesystem' and of size 55 GiB.
> > 21:03:00 DEBUG| [stdout]
> > 21:03:00 DEBUG| [stdout] Command (m for help): The partition table has been altered.
> > 21:03:00 DEBUG| [stdout] Calling ioctl() to re-read partition table.
> > 21:03:00 DEBUG| [stdout] Syncing disks.
> 
> Hi @peixiu , I think this issue is related with https://bugzilla.redhat.com/show_bug.cgi?id=1754751

it seems hit same one issue, could you add serial option for this case? without serial option has been covered by default unattend_install case, we don't need cover it again at this case.
And please add scsi-block option test for this case, it should have 2 configure for this case, one for scsi-hd, and another for scsi-block.

**Comment by YongxueHong** [comment]
> > > Tested on windows 2012-r2 failed when windows setup, hit error message as:
> > > "Windows cannot be installed to this hard disk space. This hard disk space is formatted with an unsupported version of the NTFS file system...."
> > > Check the log file found, create lvm with a "Linux filesystem" type, it should not compatibility with Windows installation used NTFS type.
> > > 21:03:00 INFO | Running 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdb'
> > > 21:03:00 DEBUG| [stdout]
> > > 21:03:00 DEBUG| [stdout] Welcome to fdisk (util-linux 2.32.1).
> > > 21:03:00 DEBUG| [stdout] Changes will remain in memory only, until you decide to write them.
> > > 21:03:00 DEBUG| [stdout] Be careful before using the write command.
> > > 21:03:00 DEBUG| [stdout]
> > > 21:03:00 DEBUG| [stdout] Device does not contain a recognized partition table.
> > > 21:03:00 DEBUG| [stdout] Created a new DOS disklabel with disk identifier 0x7a539419.
> > > 21:03:00 DEBUG| [stdout]
> > > 21:03:00 DEBUG| [stdout] Command (m for help): Created a new GPT disklabel (GUID: 352C6F1B-E8DA-854B-A8B8-DA1DAB910F23).
> > > 21:03:00 DEBUG| [stdout]
> > > 21:03:00 DEBUG| [stdout] Command (m for help): Partition number (1-128, default 1): First sector (16384-115343326, default 16384): Last sector, +sectors or +size{K,M,G,T,P} (16384-115343326, default 115343326):
> > > 21:03:00 DEBUG| [stdout] Created a new partition 1 of type 'Linux filesystem' and of size 55 GiB.
> > > 21:03:00 DEBUG| [stdout]
> > > 21:03:00 DEBUG| [stdout] Command (m for help): The partition table has been altered.
> > > 21:03:00 DEBUG| [stdout] Calling ioctl() to re-read partition table.
> > > 21:03:00 DEBUG| [stdout] Syncing disks.
> > 
> > 
> > Hi @peixiu , I think this issue is related with https://bugzilla.redhat.com/show_bug.cgi?id=1754751
> 
> it seems hit same one issue, could you add serial option for this case? without serial option has been covered by default unattend_install case, we don't need cover it again at this case.
> And please add scsi-block option test for this case, it should have 2 configure for this case, one for scsi-hd, and another for scsi-block.

Hi, @peixiu 
I see this case attaches a passthrough device (lvm), so it is fit for scsi-hd?

**Comment by peixiu** [comment]
> > > > Tested on windows 2012-r2 failed when windows setup, hit error message as:
> > > > "Windows cannot be installed to this hard disk space. This hard disk space is formatted with an unsupported version of the NTFS file system...."
> > > > Check the log file found, create lvm with a "Linux filesystem" type, it should not compatibility with Windows installation used NTFS type.
> > > > 21:03:00 INFO | Running 'echo -e "g\nn\n1\n\n\nw\n" | fdisk /dev/sdb'
> > > > 21:03:00 DEBUG| [stdout]
> > > > 21:03:00 DEBUG| [stdout] Welcome to fdisk (util-linux 2.32.1).
> > > > 21:03:00 DEBUG| [stdout] Changes will remain in memory only, until you decide to write them.
> > > > 21:03:00 DEBUG| [stdout] Be careful before using the write command.
> > > > 21:03:00 DEBUG| [stdout]
> > > > 21:03:00 DEBUG| [stdout] Device does not contain a recognized partition table.
> > > > 21:03:00 DEBUG| [stdout] Created a new DOS disklabel with disk identifier 0x7a539419.
> > > > 21:03:00 DEBUG| [stdout]
> > > > 21:03:00 DEBUG| [stdout] Command (m for help): Created a new GPT disklabel (GUID: 352C6F1B-E8DA-854B-A8B8-DA1DAB910F23).
> > > > 21:03:00 DEBUG| [stdout]
> > > > 21:03:00 DEBUG| [stdout] Command (m for help): Partition number (1-128, default 1): First sector (16384-115343326, default 16384): Last sector, +sectors or +size{K,M,G,T,P} (16384-115343326, default 115343326):
> > > > 21:03:00 DEBUG| [stdout] Created a new partition 1 of type 'Linux filesystem' and of size 55 GiB.
> > > > 21:03:00 DEBUG| [stdout]
> > > > 21:03:00 DEBUG| [stdout] Command (m for help): The partition table has been altered.
> > > > 21:03:00 DEBUG| [stdout] Calling ioctl() to re-read partition table.
> > > > 21:03:00 DEBUG| [stdout] Syncing disks.
> > > 
> > > 
> > > Hi @peixiu , I think this issue is related with https://bugzilla.redhat.com/show_bug.cgi?id=1754751
> > 
> > 
> > it seems hit same one issue, could you add serial option for this case? without serial option has been covered by default unattend_install case, we don't need cover it again at this case.
> > And please add scsi-block option test for this case, it should have 2 configure for this case, one for scsi-hd, and another for scsi-block.
> 
> Hi, @peixiu
> I see this case attaches a passthrough device (lvm), so it is fit for scsi-hd?

for drive mode, the file+scsi-hd is supported, and the file+scsi-block also is supported. 
for blockdev mode, file + scsi-hd is not supported, althrough host_device + scsi-hd is work, but I think it's invalid, host_deivce+scsi-block is correct configure.
In fact, I also think we can only test host_device+scsi-block for this case both drive mode and blockdev mode, but this need scsi qemu feature owner(qinwang) to confirm that, it's a qemu specific configure, I just can decide the virtio-win-scsi related configure, you can ask him to help confirm~

**Comment by YongxueHong** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/2472

**Comment by YongxueHong** [comment]
@peixiu  please help me to review, thanks.

**Comment by YongxueHong** [comment]
@peixiu  please patch up https://github.com/avocado-framework/avocado-vt/pull/2472 before reviewing.

File: qemu/tests/cfg/block_iscsi_lvm.cfg [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                - extra_cdrom_ks:
+                    no WinXP Win2000 Win2003 WinVista
+                    cdroms += " unattended"
+                    drive_index_unattended = 3
+                    drive_index_cd1 = 1
+            variants:
+                - cdrom:
+                    Windows:
+                        cdroms += " winutils"
+                        cd_format_winutils = scsi-cd
```
Comment (peixiu) [inline qemu/tests/cfg/block_iscsi_lvm.cfg]: Hi yhong, currently, all scsi-cd drive are connected to the same virtio-scsi-pci bus, if virtio-scsi-pci enabled dataplane, then all scsi-cd will cannot connect on due to not support,  in order to match dataplane=on situation, better here change cd_format_winutils/cd1 = ide.

**Comment by peixiu** [comment]
Tested with code on win2012-64 guest, one passed and one failed:
 (1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_hd.q35: PASS (874.37 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_block.q35: ERROR: Failures occurred while postprocess:\n\nPostprocess command: Command 'cd /usr/share/avocado-plugins-vt; lvm lvremove -ff --yes /dev/vg_test/lv_test && lvm vgremove -ff --yes vg_test && lvm pvremove -ff --yes /dev/sde1 && iscsiadm --mode node --logout -T iqn... (846.64 s)

Error message:

22:00:00 ERROR| Traceback (most recent call last):

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/core/test.py", line 955, in _run_avocado
    raise test_exception

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/core/test.py", line 855, in _run_avocado
    testMethod()

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/avocado_vt/test.py", line 301, in runTest
    raise self.__status  # pylint: disable=E0702

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/avocado_vt/test.py", line 252, in setUp
    self._runTest()

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/avocado_vt/test.py", line 452, in _runTest
    env_process.postprocess(self, params, env)

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
    return fn(*args, **kwargs)

22:00:00 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugin_vt-76.0-py3.6.egg/virttest/env_process.py", line 1724, in postprocess
    raise RuntimeError("Failures occurred while postprocess:\n%s" % err)

22:00:00 ERROR| RuntimeError: Failures occurred while postprocess:

Postprocess command: Command 'cd /usr/share/avocado-plugins-vt; lvm lvremove -ff --yes /dev/vg_test/lv_test && lvm vgremove -ff --yes vg_test && lvm pvremove -ff --yes /dev/sde1 && iscsiadm --mode node --logout -T iqn.2019-12.com.redhat:kvm-qe && iscsiadm -m node -o delete -T iqn.2019-12.com.redhat:kvm-qe --portal 127.0.0.1 && iqn=`targetcli /iscsi/ ls | grep iqn.2019-12.com.redhat:kvm-qe | awk '{print $2}'` && targetcli /iscsi/$iqn/tpg1/luns/ delete lun=0 && targetcli /iscsi/ delete $iqn && name=`targetcli /backstores/fileio/ ls | grep sys | awk '{print $2}'` && targetcli /backstores/fileio/ delete $name && mv /etc/iscsi/initiatorname.iscsi-5FR1 /etc/iscsi/initiatorname.iscsi && systemctl reset-failed iscsid.service && systemctl restart iscsid.service && rm -rf /var/lib/avocado/data/avocado-vt/images/sys && rm -rf /dev/vg_test' failed.
  stdout: b''
  stderr: b'  Volume group "vg_test" not found\n  Cannot process volume group vg_test\n'
  additional_info: None

22:00:00 ERROR| ERROR 2-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_block.q35 -> RuntimeError: Failures occurred while postprocess:

Postprocess command: Command 'cd /usr/share/avocado-plugins-vt; lvm lvremove -ff --yes /dev/vg_test/lv_test && lvm vgremove -ff --yes vg_test && lvm pvremove -ff --yes /dev/sde1 && iscsiadm --mode node --logout -T iqn.2019-12.com.redhat:kvm-qe && iscsiadm -m node -o delete -T iqn.2019-12.com.redhat:kvm-qe --portal 127.0.0.1 && iqn=`targetcli /iscsi/ ls | grep iqn.2019-12.com.redhat:kvm-qe | awk '{print $2}'` && targetcli /iscsi/$iqn/tpg1/luns/ delete lun=0 && targetcli /iscsi/ delete $iqn && name=`targetcli /backstores/fileio/ ls | grep sys | awk '{print $2}'` && targetcli /backstores/fileio/ delete $name && mv /etc/iscsi/initiatorname.iscsi-5FR1 /etc/iscsi/initiatorname.iscsi && systemctl reset-failed iscsid.service && systemctl restart iscsid.service && rm -rf /var/lib/avocado/data/avocado-vt/images/sys && rm -rf /dev/vg_test' failed.
  stdout: b''
  stderr: b'  Volume group "vg_test" not found\n  Cannot process volume group vg_test\n'
  additional_info: None
22:00:00 INFO |

File: qemu/tests/cfg/block_iscsi_lvm.cfg (hunk 1 above)
Comment (YongxueHong) [reply qemu/tests/cfg/block_iscsi_lvm.cfg]: ok , will update it.

**Comment by peixiu** [comment]
(1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_hd.q35: PASS (838.22 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_block.q35: PASS (762.66 s)

 (1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_hd.q35: PASS (1033.82 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_block.q35: PASS (1048.99 s)

 (1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_hd.q35: PASS (1001.64 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.block_iscsi_lvm.with_installation.cdrom.extra_cdrom_ks.with_scsi_block.q35: PASS (892.95 s)

LGTM~
Acked-by: Peixiu Hou phou@redhat.com

**Pull request merged**